packages/blockchain/src/blockchain.ts

```diff
@@ -581,7 +581,7 @@ export class Blockchain implements BlockchainInterface {
     }
 
     if (header.common.isActivatedEIP(4844)) {
-      const expectedExcessBlobGas = parentHeader.calcNextExcessBlobGas()
+      const expectedExcessBlobGas = parentHeader.calcNextExcessBlobGas(this.common)
```

Comment:
Hmm, but this gets awry at the fork block, right?

If we are currently at the Cancun fork but the next block is Prague, then `this.common` will point to Cancun, but the next block is Prague, so it should use the Prague parameters to calculate the excess blob gas. What is also somewhat annoying here is that the fork used obviously depends upon the timestamp of the next block. It is only certain that the next block will be a fork block if we are exactly 1 slot before the fork block.